ios/Mappers.swift

```diff
@@ -1075,6 +1075,42 @@ class Mappers {
         default: return .unknown
         }
     }
+  
+
+    class func mapToSurchargeConfiguration(from dict: [String: Any]?) throws -> SurchargeConfiguration? {
+        guard let dict = dict else { return nil }
+        guard let amount = dict["amount"] as? Int else {
+            return nil
+        }
+        
+        let surchargeConfigBuilder = SurchargeConfigurationBuilder()
+        surchargeConfigBuilder.setAmount(UInt(amount))
+        
+        let consentBuilder = SurchargeConsentBuilder()
+        if let consentDict = dict["consent"] as? [String: Any] {
+            if let collectionString = consentDict["collection"] as? String {
+                let collection: SurchargeConsentCollection
+                switch collectionString {
```

Comment:
worth being case insensitive? 

```suggestion
                switch collectionString.lowercased() {
```

Reply:
Yes, we can have it as a protection. Updated!
The string is from the RN side, and in the RN code base, the consent is an enum, so this should be fine.